pkg/gadgets/run/types/types.go

```diff
@@ -85,10 +94,56 @@ type Printer interface {
 	Logf(severity logger.Level, fmt string, params ...any)
 }
 
+// Type represents the type of a field.
+
+type Kind uint8
+
+// TODO: use reflect.Kind? (once we resolve TODO below)
+const (
+	KindNone = iota
+	KindUint8
+	KindUint16
+	KindUint32
+	KindUint64
+	KindInt8
+	KindInt16
+	KindInt32
+	KindInt64
+	KindBool
+	KindFloat32
+	KindFloat64
+	KindString
+	KindArray
+
+	// TODO: this should be made more generic to avoid coupling operators with this
+	KindL3Endpoint
+	KindL4Endpoint
+	KindTimestamp
+)
+
+type Type struct {
+	Kind Kind
+
+	// Fields only for KindArray
+	ArrayNElements int // number of elements in the array
+	ArrayType      *Type
+}
+
+// ColumnDesc describes how a column is built. It's basically a serializable version of
+// columns.DynamicField
+type ColumnDesc struct {
+	Name      string
+	BlobIndex int // -1: virtual, 0: ebpf, 1: fixed length, 1+ strings
+	Type      Type
+	Offset    uintptr
+}
+
 type GadgetInfo struct {
 	GadgetMetadata *GadgetMetadata
+	Columns        []ColumnDesc
 	ProgContent    []byte
 	GadgetType     gadgets.GadgetType
+	EventFactory   *EventFactory
```

Comment:
Since the `type GadgetInfo` is used in the [GetGadgetInfo gRPC method](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/gadget-service/api/api.proto#L94), I think there should be some documentation about the that in:

https://inspektor-gadget.io/docs/latest/reference/grpc/

Even if it is just a few sentences explaining how the json is encoded in the gRPC field. For now, the source says:
```
message GetGadgetInfoResponse {
  bytes info = 1; // encoded in json
}
```
But it does not say what is encoded in json and how to find the Golang struct that is marshalled into json.

Reply:
Added a commit with few sentences about it.